src/shared/sam/debugger/awsSamDebugConfiguration.ts

```diff
@@ -37,10 +37,10 @@ export function isCodeTargetProperties(props: TargetProperties): props is CodeTa
     return props.target === CODE_TARGET_TYPE
 }
 
-export function createAwsSamDebugConfig(
+export function createTemplateAwsSamDebugConfig(
```

Comment:
a docstring would be useful, then the verbose method name can be avoided. If a "lambda" variant is added later, it would make sense to add that as a parameter to this function, rather than have a new function. So we can avoid over-specifying the function purpose meanwhile.

Reply:
I'm overspecifying because this one is specific to TEMPLATE_TARGET_TYPE instead of CODE_TARGET_TYPE. Would you rather have one function that branches into both of those with a more generic name or two separate functions?

Reply:
if it were up to me I would branch. Helps with discovery when we group similar things together instead of spreading them out.

Reply:
I'll give it a look. The added params are pretty different (at least between `code` and `template`, maybe not so much if we move remote Lambda invokes to this) so we may need to use some ugly params objects as the function args.